patches/google-cloud-bigquery-JobWaitTrait.php.patch

```diff
@@ -1,12 +1,18 @@
 --- src/JobWaitTrait.php	2023-11-01 13:07:13.826212516 +0100
 +++ src/JobWaitTrait.php	2023-11-01 13:07:24.542742533 +0100
-@@ -55,7 +55,9 @@
+@@ -55,7 +55,15 @@
                  }
              };
 
 -            (new ExponentialBackoff($maxRetries))
 +            (new ExponentialBackoff($maxRetries, function (\Throwable $e) {
-+                return $e->getCode() !== 499;
++                if ($e->getCode() === 499) {
++                    return false;
++                }
++                if (str_contains($e->getMessage(), 'Job did not complete within the allowed number of retries.')) {
```

Comment:
Tohle bych sem nedaval, sestreli to jakykoliv dele bezici job. Postupne bych asi pridava response codes pro ktere skonci. Pridal bych 400 pro:
<img width="753" alt="image" src="https://github.com/keboola/google-bigquery-transformation/assets/903531/d64f80e8-7a2b-4ce1-bf26-d4e76be842c5">

Reply:
Jo zkůšal sem to lokálně a je to na prd takto, přidal sem tam přímo check na `reached the maximum number of iterations` což správně shodí tu query po minutě kdy skončí v BQ.

Reply:
Nevim jestli nebude lepsi pridat ten kod `400` nez tu message?

Reply:
Mohlo by to pokryt vice veci s tim kodem, zas je otazka jestli by to pak nekonvertovalo na user error neco co ma byt app error...

Reply:
Přemýšlal sem nad tým tam hodit tu 400.
Nemusíš sa bát, že by to konvertovalo něco špatně na UserException protože to konvertuje všecko na UserException :)
https://github.com/keboola/google-bigquery-transformation/blob/5388f9a4bfd69477a41091c07f540019eeaa156e/src/Transformation.php#L172-L183

Reply:
Tady mi to s verzi `1.5.3-dev-1` hodil internal error https://keboola.atlassian.net/wiki/spaces/ENGG/pages/3100147728/BigQuery+-+Another+job+is+currently+running+in+the+session

Reply:
:thinking: to není chyba z transformace bych řekl, ale chyba z connection, ne? Vypadá to jako failnutý drop workspace.

Reply:
nn z transformace

Reply:
jo sorry je to tak. Ono to totiž nepadlo v transformačním kódu, ale v 
https://github.com/keboola/google-bigquery-transformation/blob/5388f9a4bfd69477a41091c07f540019eeaa156e/src/Transformation.php#L247-L252
Ale aj tam by to mělo dělat retry.